### PR TITLE
Remove the "enableFeatureRemoveSearchTools" feature flag

### LINKED ISSUE #9112: Remove the `enableFeatureRemoveSearchTools` feature flag
We enabled this feature in all envs in https://github.com/mozilla/addons-frontend/issues/9110, so we can safely remove the feature flag (and the if/else code) after today's push.

We want to remove the config for `enableFeatureRemoveSearchTools` as well as all the code that looks like `if (_config.get('enableFeatureRemoveSearchTools')) {`, etc.

### PR DESCRIPTION
**Pull request by atiqueahmedziad**
Fixes #9112 

Removed the codes related to "enableFeatureRemoveSearchTools"

### CODE REVIEW

File: src/amo/pages/SearchTools/index.js [hunk 1]
```diff
@@ -29,41 +19,10 @@ type Props = {|
...
 export class SearchToolsBase extends React.Component<InternalProps> {
-  static defaultProps = {
-    _config: config,
-  };
-
-  constructor(props: InternalProps) {
-    super(props);
-
-    // See: https://github.com/mozilla/addons-frontend/issues/8679
-    if (props._config.get('enableFeatureRemoveSearchTools')) {
```
Comment (willdurand) [inline src/amo/pages/SearchTools/index.js]: We need to keep the logic inside the `if` block now. This is a redirect to support old URLs.

File: tests/unit/amo/pages/TestSearchTools.js [hunk 2]
```diff
@@ -55,33 +53,4 @@ describe(__filename, () => {
 
     expect(root.find(HeadLinks)).toHaveLength(1);
   });
-
-  describe('enableFeatureRemoveSearchTools = true', () => {
-    it('sends a server redirect', () => {
```
Comment (willdurand) [inline tests/unit/amo/pages/TestSearchTools.js]: We need to keep this test case all the time now. I suppose you can remove the `describe` block above, though, and only keep/rename `it('sends a server redirect to support old search tool URLs')`

File: tests/unit/amo/pages/TestSearchTools.js [hunk 3]
```diff
@@ -55,33 +53,4 @@ describe(__filename, () => {
...
-      const _config = getFakeConfig({ enableFeatureRemoveSearchTools: true });
```
Comment (atiqueahmedziad) [inline tests/unit/amo/pages/TestSearchTools.js]: Will it be as it is now ? or what should we set to `_config` here ?

Comment (willdurand) [reply tests/unit/amo/pages/TestSearchTools.js]: no need to pass a `_config` object, it is by default now.

File: src/amo/components/SectionLinks/index.js [hunk 4]
```diff
@@ -90,17 +82,6 @@ export class SectionLinksBase extends React.Component<InternalProps> {
       );
     }
 
-    // See: https://github.com/mozilla/addons-frontend/issues/8680
-    if (!_config.get('enableFeatureRemoveSearchTools')) {
```
Comment (atiqueahmedziad) [inline src/amo/components/SectionLinks/index.js]: @willdurand do we need the codes below this ?

File: tests/unit/amo/pages/TestSearchTools.js (hunk 2 above)
Comment (atiqueahmedziad) [reply tests/unit/amo/pages/TestSearchTools.js]: @willdurand Will I do the same for other tests above ? Or just only this one ?

File: src/amo/components/SectionLinks/index.js (hunk 4 above)
Comment (willdurand) [reply src/amo/components/SectionLinks/index.js]: no, only code when the condition checks that `enableFeatureRemoveSearchTools` is `true`.

**Comment by atiqueahmedziad** [comment]
Updated. Please check if it is alright.

File: tests/unit/amo/pages/TestSearchTools.js [hunk 5]
```diff
@@ -56,32 +55,24 @@ describe(__filename, () => {
...
-      const _config = getFakeConfig({ enableFeatureRemoveSearchTools: false });
-      const fakeDispatch = sinon.spy(store, 'dispatch');
+  it('does not send a server redirect to support old search tool URLs', () => {
+    const fakeDispatch = sinon.spy(store, 'dispatch');
 
-      render({ _config });
-
-      sinon.assert.notCalled(fakeDispatch);
-    });
+    sinon.assert.notCalled(fakeDispatch);
```
Comment (willdurand) [inline tests/unit/amo/pages/TestSearchTools.js]: This isn't needed I believe.

**Comment by willdurand** [comment]
thanks @atiqueahmedziad!

**Pull request merged**